### PR TITLE
Relative file names for grep source

### PR DESCRIPTION
**Pull request by valff**
Current implementation passes full directory path as target to grep command. So, for example, ag command returns full file paths too.

### CODE REVIEW

File: autoload/unite/helper.vim [hunk 1]
```diff
@@ -546,10 +546,18 @@ function! unite#helper#is_prompt(line) "{{{
         \ || (context.prompt_direction !=# 'below' && a:line <= prompt_linenr)
 endfunction"}}}
 
+function! unite#helper#relative_target(target) "{{{
+  let target = fnamemodify(a:target, ':.')
+  if target == getcwd()
+    return '.'
+  endif
+  return target
```
Comment (Shougo) [inline autoload/unite/helper.vim]: OK.  But please use `unite#util#substitute_path_separator()` for `target`.

Comment (Shougo) [inline autoload/unite/helper.vim]: I think `substitute(target, '/$', '', '')` also should be add it.

File: autoload/unite/helper.vim [hunk 2]
```diff
@@ -546,10 +546,18 @@ function! unite#helper#is_prompt(line) "{{{
         \ || (context.prompt_direction !=# 'below' && a:line <= prompt_linenr)
 endfunction"}}}
 
+function! unite#helper#relative_target(target) "{{{
+  let target = fnamemodify(a:target, ':.')
```
Comment (Shougo) [inline autoload/unite/helper.vim]: `let target = unite#util#substitute_path_separator(fnamemodify(a:target, ':.'))` is better.

File: autoload/unite/helper.vim [hunk 3]
```diff
@@ -546,10 +546,18 @@ function! unite#helper#is_prompt(line) "{{{
...
+  if target == getcwd()
```
Comment (Shougo) [inline autoload/unite/helper.vim]: `if target == unite#util#substitute_path_separator(getcwd())` is better.

**Comment by Shougo** [comment]
OK.  I reviewed it.

File: autoload/unite/helper.vim (hunk 1 above)
Comment (Shougo) [inline autoload/unite/helper.vim]: Oh, `substitute(target, '[^:]zs/$', '', '')` is better.

**Comment by valff** [comment]
Fixed.

**Comment by Shougo** [comment]
Thanks.

**Pull request merged**